rpc/client.go

```diff
@@ -213,9 +213,32 @@ func (c *Client) getClientUsingCache(chainID uint64) (chain.ClientInterface, err
 	return client, nil
 }
 
+func getProviderPriorityByURL(url string) int {
+	if strings.Contains(url, "/nodefleet/") || strings.Contains(url, "anvil") {
+		return 0
+	} else if strings.Contains(url, "/infura/") {
+		return 1
+	} else if strings.Contains(url, "infura.io/") {
+		return 2
+	} else if strings.Contains(url, "/grove/") {
+		return 3
+	}
+
+	return 4
+}
+
 func (c *Client) getEthClients(network *params.Network) []ethclient.RPSLimitedEthClientInterface {
+	// Keys are defining the order of providers being used
```

Comment:
I see 2 problems:
- if the same key priority is used 
- some code duplication

```
type Provider struct {
	Key      string
	URL      string
	Auth     string
	Priority int
}

func createProvider(key, url, credentials string, providers *[]Provider) {
	priority := getProviderPriorityByURL(url)
	*providers = append(*providers, Provider{
		Key:      key,
		URL:      url,
		Auth:     credentials,
		Priority: priority,
	})
}

func (c *Client) prepareProviders(network *params.Network) []Provider {
	var providers []Provider

	// Retrieve the proxy provider configuration
	proxyProvider, err := getProviderConfig(c.providerConfigs, ProviderStatusProxy)
	if err != nil {
		c.log.Warn("could not find provider config for status-proxy", "error", err)
	}

	// Add main and fallback providers
	createProvider(ProviderMain, network.RPCURL, "", &providers)
	createProvider(ProviderFallback, network.FallbackURL, "", &providers)

	// If the proxy provider is enabled, add it and its fallback options
	if proxyProvider.Enabled {
		credentials := proxyProvider.User + ":" + proxyProvider.Password
		createProvider(ProviderStatusProxy, network.DefaultRPCURL, credentials, &providers)
		createProvider(ProviderStatusProxyFallback, network.DefaultFallbackURL, credentials, &providers)
		createProvider(ProviderStatusProxyFallback2, network.DefaultFallbackURL2, credentials, &providers)
	}

	// Sort providers by priority
	sort.Slice(providers, func(i, j int) bool {
		return providers[i].Priority < providers[j].Priority
	})

	return providers
}
```

and use it in `getEthClients`